.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build binaries
+
+on:
+  pull_request:
+    branches:
+    - master
+  push:
+    tags:
+    - "*"
```

Comment:
Follow-up insights:

- This doesn't trigger a build when being merging into `master`. But it should trigger and create build artifacts.
- Those build artifacts then should be reused by a "Publish" or "Release" workflow up pushing a Git tag (i.e. as configured in the lines above).